config/lura.go

```diff
@@ -96,3 +97,31 @@ func FromLura(srvCfg config.ServiceConfig) (*Config, error) {
 	}
 	return cfg, nil
 }
+
+// MemoizedConfigParser creates a function that parses a lura config,
+// using a ConfigParserFn and memoizes its results in a thread safe way.
+func MemoizedConfigParser(cfgParser ConfigParserFn) ConfigParserFn {
+	var mutex sync.RWMutex
+	var cfg *Config
+	var err error
+
+	return func(srvCfg luraconfig.ServiceConfig) (*Config, error) {
```

Comment:
It's easier to use `sync.Once`, but if we are going to use just a single input, this can be done at the constructor (as in krakend-opencensus and krakend-metrics)